### ISSUE TITLE
Piping data changes workfolder to temp

### ISSUE THREAD

**Issue by klonuo** [wkhtmltopdf/wkhtmltopdf#3014]
Please take a look at this session:

```
C:\temp\test>dir
 Volume in drive C has no label.
 Volume Serial Number is 0E52-CE22

 Directory of C:\temp\test

21.06.2016  14:19    <DIR>          .
21.06.2016  14:19    <DIR>          ..
20.06.2016  11:06         1,461,186 screenshot.png
21.06.2016  14:18               105 test.html
               2 File(s)      1,461,291 bytes
               2 Dir(s)  252,166,000,640 bytes free

C:\temp\test>type test.html
<figure>

<img src="screenshot.png" alt="Screenshot" /><figcaption>Screenshot</figcaption>

</figure>
C:\temp\test>type test.html | wkhtmltoimage - test.png
Loading page (1/2)
Warning: Failed to load file:///C:/Users/klonuo/AppData/Local/Temp/screenshot.png (ignore)
Rendering (2/2)
Done

C:\temp\test>
```

So when piping data, workfolder is changed to %tmp% and wkhtmltoimage cant locate referenced files from html input.

**Comment by klonuo** [comment]
Just to add that I tried also: `--cache-dir c:\temp\test` and `--allow c:\temp\test` w/o success

**Comment by klonuo** [comment]
Further, setting %tmp% to current working folder, reveals that when using stdin pipe, wkhtmltoimage creates temp file, so there is no point in using stdin

Setting %tmp% to nonexisting folder crashes wkhtmltoimage

**Comment by artheus** [comment]
I :+1: on this issue.

There are use-cases where piping html to wkhtmltopdf with relative links makes sense. I am currently in one of those. So the only solution for me is to create the temp files myself in the directory with the static media resources in them. This is not optimal for me, as it makes the media resource directory cluttered with files that are not supposed to be there.

So my suggestion is that wkhtmltopdf should allow piping to stdin, keep the current work directory, and in the background create html-files in the system temp directory.

Or! allow sending a command line parameter telling where to find relative files.

**Comment by suushiemaniac** [comment]
Bumping this issue as I have recently run into the same exact scenario as described by @artheus.
My workaround also consists in deliberately cluttering the static resources directory with intermediary HTML files.

I would like to see a CLI option that tells `wkhtmltopdf` where to find relative files.

@alloylab Why have you added the "NeedInfo" label, and what kind of information do you require?

**Comment by ashkulz** [comment]
Does this happen only on Windows? There's no code which changes the working directory, so I assume it's something specific to that OS.

**Comment by suushiemaniac** [comment]
This actually happens to me under Linux, more specifically manjaro 18.1.3
Kernel 5.4.0
wkhtmltopdf 0.12.5 (with patched qt)

**Comment by ashkulz** [comment]
@suushiemaniac that's weird, are the reproduction instructions the same on Linux?

**Comment by suushiemaniac** [comment]
Yes indeed, just replace the `cmd` commands with their `bash` equivalents and I get a very similar error:

```bash
$ ls
myimage.png  test.html

$ cat test.html 
<figure>
<img src="myimage.png" alt="Screenshot" /><figcaption>Screenshot</figcaption>
</figure>

$ cat test.html | wkhtmltopdf - test.pdf
Loading pages (1/6)
Warning: Failed to load file:///tmp/myimage.png (ignore)          
Counting pages (2/6)                                               
Resolving links (4/6)                                                       
Loading headers and footers (5/6)                                           
Printing pages (6/6)
Done
```

**Comment by SebastiaanKlippert** [comment]
Hi @ashkulz do you perhaps have any update on this?
I have had three people report this in the past week in my Go wrapper which commonly uses `stdin` as input.
The issue can be replicated in `0.12.5` in both Windows and Linux where it always looks for files relative to the TEMP path. 
Just running `cat test.html | wkhtmltopdf - test.pdf` will reproduce when using a local resource in the HTML, running `wkhtmltopdf test.html test.pdf` will work fine with the same code.
Described my findings here https://github.com/SebastiaanKlippert/go-wkhtmltopdf/issues/28#issuecomment-585308645 

I guess it is a Qt thing, you mention there is no code in wkhtmltopdf to set the working dir, but it might have something to do with a difference in calling Qt where setting a web Base URL behaves differently than using stdin. My C++ knowledge is too limited to understand all code, but I would also expect there to be a possibilty to change this behavior if needed. Looks like Qt defaults to TEMP if no base URL is not given, if we could somehow pass it the working dir we should be there.

**Comment by andreasph** [comment]
We are experiencing the same problem on macos and debian buster based docker container
our command can be narrowed down to:
```
echo "<img src=\"https://www.google.com/images/branding/googlelogo/2x/googlelogo_color_272x92dp.png\">" | /usr/local/bin/wkhtmltoimage --quality 100 --format jpeg --width 150 --height 350 - -
```
the real src url is not relevant
we run this with `exec.Command()` in golang with pipeping (so there are two separated command, the `echo` one and the one for `wkhtmltoimage`, that communicates through `exec.Cmd.StdoutPipe()`)

I can see that wkhtmltoimage try to access `file:///https://www.google.com/images/branding/googlelogo/2x/googlelogo_color_272x92dp.png`, prepending the `file` schema to the url. this doesn't happen if I run the command directly from cli in a terminal

we ended up saving the html in a tempfile and running `/usr/local/bin/wkhtmltoimage --quality 100 --format jpeg --width 150 --height 350 /tmp/tmpfile.html -`
